grai-server/app/Dockerfile

```diff
@@ -26,6 +29,8 @@ RUN apt update \
     curl \
     && rm -rf /var/lib/apt/lists/*
 
+RUN curl https://sh.rustup.rs -sSf | bash -s -- -y
```

Comment:
I may be missing it but where is the rust dependency coming from? I checked to see if celery or redis had a dependency but couldn't find anything listed (although there is a rust implementation for celery).

Reply:
This was being caused by adding `grai-source-postgres`, I suspect somewhere there is a dependency that hasn't been built for arm64, and hence my machine is falling back to building from source.  I agree it would be good to remove this if we can, alternatively we could look at a two step build.

Reply:
Ahh it's actually being smuggled through in via the orjson dependency in `grai-client`. It took me a bit to figure out why poetry was attempting to build from source - the problem ended up being with the lock file which we've been creating using our MacOS dev machines. The fix is to re-create the lock file when building the docker image at the poetry install step.


```
RUN poetry config virtualenvs.create false  \
    && poetry lock \
    && poetry install --no-interaction --no-ansi --no-root --only main \
    && rm -rf ~/.cache/pypoetry/cache \
    && rm -rf ~/.cache/pypoetry/artifacts
```

Reply:
I haven't entirely figured out why this is causing an issue. The lock files are supposed to be portable but something about orjson is behaving differently.

Reply:
Strange, I have used `poetry lock --no-update` which seems to work.